app/controllers/course/forum/topics_controller.rb

```diff
@@ -9,8 +9,10 @@ class Course::Forum::TopicsController < Course::Forum::ComponentController
   before_action :add_topic_breadcrumb
 
   def show
-    @posts = @topic.posts
+    @posts = @topic.posts.order(created_at: :asc).includes(:creator).to_a
     @topic.views.create(user: current_user)
+    @post = @topic.acting_as.posts.build(title: t('course.discussion.posts.reply_title',
```

Comment:
potential problem means the problem you can not foresee. You told me that `acting_as` gem has a lot of bugs right? Then I think building post from discussion topic is the most correct thing to do. And I can not foresee what kind of problems will be caused if we use concrete topic all the time.